rosidl_typesupport_opendds_cpp/resource/msg__type_support.cpp.em

```diff
@@ -111,55 +107,47 @@ convert_ros_message_to_dds(
     size_t size = @(member.type.size);
 @[    else]@
     size_t size = ros_message.@(member.name).size();
-    if (size > (std::numeric_limits<DDS_Long>::max)()) {
+    if (size > (std::numeric_limits<int>::max)()) {
       throw std::runtime_error("array size exceeds maximum DDS sequence size");
     }
 @[      if isinstance(member.type, BoundedSequence)]@
     if (size > @(member.type.maximum_size)) {
       throw std::runtime_error("array size exceeds upper bound");
     }
 @[      end if]@
-    DDS_Long length = static_cast<DDS_Long>(size);
-    if (length > dds_message.@(member.name)_.maximum()) {
-      if (!dds_message.@(member.name)_.maximum(length)) {
+    if (size > dds_message.@(member.name)_().max_size()) {
         throw std::runtime_error("failed to set maximum of sequence");
-      }
-    }
-    if (!dds_message.@(member.name)_.length(length)) {
-      throw std::runtime_error("failed to set length of sequence");
     }
+    dds_message.@(member.name)_().resize(size);
 @[    end if]@
     for (size_t i = 0; i < size; i++) {
 @[    if isinstance(member.type.value_type, AbstractString)]@
-      DDS_String_free(dds_message.@(member.name)_[static_cast<DDS_Long>(i)]);
-      dds_message.@(member.name)_[static_cast<DDS_Long>(i)] =
-        DDS_String_dup(ros_message.@(member.name)[i].c_str());
+      dds_message.@(member.name)_()[i] =
+        ros_message.@(member.name)[i].c_str();
```

Comment:
Can we use std::string instead of C strings?

Reply:
From what I understand by looking at the generated classes, probably no.
rosidl string is std::basic_string for char type:

  // field types and members
  using _data_type =
    std::basic_string<char, std::char_traits<char>, typename ContainerAllocator::template rebind<char>::other>;
  _data_type data;

opendds string is std::string, so need to use c_str() for assignment